Utils/Functions/Filesystem/Extract/script.js

```diff
@@ -1,144 +1,169 @@
 include("utils.functions.filesystem.files");
 
-/**
-* CabExtract prototype
-* @constructor
-*/
-function CabExtract() {
-}
+const ProcessBuilderClass = Java.type("java.lang.ProcessBuilder");
+const FileClass = Java.type("java.io.File");
 
 /**
-* sets wizard
-* @param {SetupWizard} wizard setup wizard
-* @returns {CabExtract} CabExtract object
-*/
-CabExtract.prototype.wizard = function (wizard) {
-    this._wizard = wizard;
-    return this;
-}
+ * CabExtract class
+ */
+// eslint-disable-next-line no-unused-vars
+class CabExtract {
+    constructor() {
+        // do nothing
+    }
 
-/**
-* sets archive
-* @param {string} archive archive which shall be extracted
-* @returns {CabExtract} CabExtract object
-*/
-CabExtract.prototype.archive = function (archive) {
-    this._archive = archive;
-    return this;
-}
+    /**
+     * Sets the setup wizard
+     *
+     * @param {SetupWizard} wizard The setup wizard
+     * @returns {CabExtract} The CabExtract object
+     */
+    wizard(wizard) {
+        this._wizard = wizard;
+        return this;
+    }
 
-/**
-* sets progress message text
-* @param {string} progressMessage progress message
-* @returns {CabExtract} CabExtract object
-*/
-CabExtract.prototype.message = function (progressMessage) {
-    this._progressMessage = progressMessage;
-    return this;
-}
+    /**
+     * Sets the CAB archive
+     *
+     * @param {string} archive The CAB archive which shall be extracted
+     * @returns {CabExtract} The CabExtract object
+     */
+    archive(archive) {
+        this._archive = archive;
+        return this;
+    }
 
-/**
-* sets destination
-* @param {string} destination place where the archive shall be extracted
-* @returns {CabExtract} CabExtract object
-*/
-CabExtract.prototype.to = function (destination) {
-    this._destination = destination;
-    return this;
+    /**
+     * Sets the progress message text
+     *
+     * @param {string} progressMessage The progress message
+     * @returns {CabExtract} The CabExtract object
+     */
+    message(progressMessage) {
+        this._progressMessage = progressMessage;
+        return this;
+    }
+
+    /**
+     * Sets the destination for the extracted archive content
+     *
+     * @param {string} destination The place where the archive shall be extracted
+     * @returns {CabExtract} The CabExtract object
+     */
+    to(destination) {
+        this._destination = destination;
+        return this;
+    }
+
+    /**
+     * Extracts the archive to the previously set destination
+     *
+     * @param {string} args The arguments for the extraction
+     * @returns {void}
+     */
+    extract(args) {
+        if (this._wizard) {
+            let progressMessage = this._progressMessage;
+            if (!progressMessage) {
+                progressMessage = tr("Please wait while {0} is extracted...", this._archive);
+            }
+
+            this._wizard.progressBar(progressMessage);
```

Comment:
@plata @ImperatorS79 @qparis is this progress bar ever shown or updated?

Reply:
Doesn't look like on the first glance...

Reply:
Should I delete it?

Reply:
No I was getting the "extracting file" progress bar.

Reply:
@Zemogiter does the progressbar do anything or is it always shown empty?
If it is always empty it may be a good idea to replace it with an info message in the wizard.

Reply:
The progress bar animation works.

Reply:
What do you mean by works?

According to my understanding of the code the progressbar should always show a progress of 0%, because it is never updated. In comparison take a look at https://github.com/PhoenicisOrg/scripts/pull/1047/files#diff-373a99ad00ee4b6509369947c33fc266R167 in the `Extractor` class, here the progressbar is updated during the `uncompress` call.